### PR TITLE
fixed #2441

### LINKED ISSUE #2441: Grunt, build & changes to libraries.
### Subject of the issue/enhancement/features
Start `grunt dev`, then make a change to a file in core/js/libraries. Grunt will detect the change and start a build process, but the corresponding core/js/libraries file in the build folder will be unchanged.

In order to get the change into the build folder, you must manually stop then restart `grunt dev`.

Is this by design, or should libraries get updated on change like other files in src/?

### PR DESCRIPTION
**Pull request by NayanKhedkar**
Fixed #2441 and also changes the calling of copy task now it's only allowed to copying modified files.

### CODE REVIEW

File: grunt/config/watch.js [hunk 1]
```diff
@@ -33,62 +33,66 @@ module.exports = {
...
     },
     required: {
-        files: ['<%= sourcedir %>extensions/*/required/**/*', '<%= sourcedir %>extensions/*/libraries/**/*'],
-        tasks: 'copy:required'
+        files: [
+            '<%= sourcedir %>extensions/*/required/**/*',
+            '<%= sourcedir %>extensions/*/libraries/**/*',
+            '<%= sourcedir %>components/**/libraries/**/*',
+            '<%= sourcedir %>menu/<%= menu %>/libraries/**/*',
+            '<%= sourcedir %>theme/<%= theme %>/libraries/**/*'
```
Comment (oliverfoster) [inline grunt/config/watch.js]: these `libraries` directives should be `required` directives

File: grunt/config/watch.js [hunk 2]
```diff
@@ -33,62 +33,66 @@ module.exports = {
...
+        tasks: ['newer:copy:scriptLoader']
     },
     libraries: {
-        files: [
-            '<%= sourcedir %>core/js/libraries/require.min.js',
-            '<%= sourcedir %>core/js/libraries/modernizr.js',
-            '<%= sourcedir %>core/js/libraries/jquery.min.js'
-        ],
-        tasks: ['copy:libraries']
+        files: ['<%= sourcedir %>core/js/libraries/**/*'],
```
Comment (oliverfoster) [inline grunt/config/watch.js]: this bit should include the `libraries` folder directives for extensions, menus, components etc

File: grunt/config/watch.js [hunk 3]
```diff
@@ -18,11 +18,11 @@ module.exports = {
     },
     courseJson: {
         files: ['<%= sourcedir %>course/**/*.<%= jsonext %>'],
-        tasks : ['jsonlint', 'check-json', 'copy:courseJson', 'schema-defaults', 'create-json-config']
+        tasks : ['jsonlint', 'check-json', 'newer:copy:courseJson', 'schema-defaults', 'create-json-config']
```
Comment (oliverfoster) [inline grunt/config/watch.js]: did the inclusion of the `newer:` library give any benefits?

Comment (NayanKhedkar) [reply grunt/config/watch.js]: Yes, it facilitates to copy only modified files instead of all the files.

Comment (oliverfoster) [reply grunt/config/watch.js]: is it faster?

File: grunt/config/watch.js (hunk 2 above)
Comment (NayanKhedkar) [reply grunt/config/watch.js]: I have included the reference of the directive  for extensions, menus, components into required directives, here is the reference,

https://github.com/adaptlearning/adapt_framework/blob/b6988c335cd9f8b077c7d9f52fc379a5ddc6721b/grunt/config/copy.js#L211

File: grunt/config/watch.js (hunk 3 above)
Comment (NayanKhedkar) [reply grunt/config/watch.js]: yes, as compared to the previous one.

File: grunt/config/watch.js (hunk 2 above)
Comment (oliverfoster) [reply grunt/config/watch.js]: the `required` folder and the `libraries` folder are two different things.

File: grunt/config/watch.js (hunk 3 above)
Comment (oliverfoster) [reply grunt/config/watch.js]: it looks as though newer only compares file modified times one way 
https://github.com/tschaub/grunt-newer/blob/master/lib/util.js#L27-L28

this means it probably won't work if i paste an older file over a newer file?

Comment (NayanKhedkar) [reply grunt/config/watch.js]: If the file in build folder the same as the file in src folder then it doesn't copy to build folder.when you paste an older file over a newer file. watch task listens to the changes and calls newer:copy task and it works in the above case also.

File: grunt/config/watch.js (hunk 2 above)
Comment (NayanKhedkar) [reply grunt/config/watch.js]: Do we really need 

> 

{
                    expand: true,
                    src: ['components/**/libraries/**/*', 'extensions/**/libraries/**/*', 'menu/<%= menu %>/libraries/**/*', 'theme/<%= theme %>/libraries/**/*'],
                    cwd: '<%= sourcedir %>',
                    dest: '<%= outputdir %>/libraries/',
                    filter: function(filepath) {
                        return grunt.config('helpers').includedFilter(filepath);
                    },
                    rename: _.partial(collate, "libraries")
                }

> at  https://github.com/adaptlearning/adapt_framework/blob/b6988c335cd9f8b077c7d9f52fc379a5ddc6721b/grunt/config/copy.js#L212

so can we move this to 
https://github.com/adaptlearning/adapt_framework/blob/b6988c335cd9f8b077c7d9f52fc379a5ddc6721b/grunt/config/copy.js#L199

Comment (oliverfoster) [reply grunt/config/watch.js]: Yup, that works.

Comment (NayanKhedkar) [reply grunt/config/watch.js]: yes, I will do the change an update it soon.

**Pull request merged**